src/Compilers/CSharp/Portable/Symbols/AssemblySymbol.cs

```diff
@@ -1046,15 +926,22 @@ bool IAssemblySymbol.GivesAccessTo(IAssemblySymbol assemblyWantingAccess)
             }
 
             var assembly = assemblyWantingAccess as AssemblySymbol;
-            if (assembly == null)
+
+            var assemblyWantingAccessName = assembly?.Name ?? assemblyWantingAccess.Identity.Name;
+            var asemblyWantingAccessKey = assembly?.PublicKey ?? assemblyWantingAccess.Identity.PublicKey;
```

Comment:
why use assembly at all in these two lines?  since you can just get this data off of assemblyWantingAccess...?  If there is a good reason, then can you add a comment explaining as such.

Reply:
Getting it from `assemblyWantingAccess` would require getting its `Identity`, which would cause binding of attributes (attributes contain, e.g., the public key). This is required to avoid an infinite recursive cycle in binding attributes during compilation.

Reply:
That would be a great thing to mention in the code :)  Any code that helps with infinite recursion would be good to call out :)